core/src/main/scala/cats/data/EitherT.scala

```diff
@@ -96,6 +96,18 @@ final case class EitherT[F[_], A, B](value: F[Either[A, B]]) {
 
   def leftMap[C](f: A => C)(implicit F: Functor[F]): EitherT[F, C, B] = bimap(f, identity)
 
+  def leftFlatMap[BB >: B, D](f: A => EitherT[F, D, BB])(implicit F: Monad[F]): EitherT[F, D, BB] =
+    EitherT(F.flatMap(value) {
+      case Left(a) => f(a).value
+      case r@Right(_) => F.pure(r.leftCast)
```

Comment:
Tiny nitpick, but can we change this to `case r: Right => ...` since we don't actually want to extract anything?

Reply:
While in practice I don't think it matters, the use of `:` implies type casing which is generally used in contexts where parametricity is violated: https://typelevel.org/blog/2014/11/10/why_is_adt_pattern_matching_allowed.html

I could have sworn there were some cases where `:` was allowed and ADT patmat wasn't but I can't remember.

Reply:
Very interesting article, thanks a lot! :) 
I always thought that scalac transforms `case t: Type` into an `isInstanceOf` check whereas `case Type(_)` would be turned into `Type.unapply(...)`, but that article shows that my assumption was false. 
It seems then, there's no real benefit to my suggestion, so disregard.